File: src/main/kotlin/magicbook/gtlitecore/core/module/GTLiteModules.java
package magicbook.gtlitecore.core.module;

import magicbook.gtlitecore.api.module.IModuleContainer;
import magicbook.gtlitecore.api.module.ModuleContainer;
import magicbook.gtlitecore.api.utils.GTLiteValues;
import org.jetbrains.annotations.ApiStatus;

@ApiStatus.Internal
@ModuleContainer
public class GTLiteModules implements IModuleContainer
{

    public static final String MODULE_CORE = "core";
    public static final String MODULE_INTEGRATION = "integration";
    public static final String MODULE_WORLDGEN = "worldgen";

    public static final String MODULE_JEI = "jei";
    public static final String MODULE_TOP = "top";
    public static final String MODULE_GRS = "grs";

    /**
     * {@inheritDoc}
     */
    @Override
    public String getId()
    {
        return GTLiteValues.MODID;
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/core/module/GTLiteModules.kt
package gregtechlite.gtlitecore.core.module

import gregtechlite.gtlitecore.api.MOD_ID
import gregtechlite.gtlitecore.api.module.IModuleContainer
import gregtechlite.gtlitecore.api.module.ModuleContainer

@ModuleContainer
internal class GTLiteModules : IModuleContainer
{

    companion object
    {

        const val MODULE_CORE: String = "core"
        const val MODULE_INTEGRATION: String = "integration"
        const val MODULE_WORLDGEN: String = "worldgen"
        const val MODULE_JEI: String = "jei"
        const val MODULE_TOP: String = "top"
        const val MODULE_AE2: String = "appeng"

    }

    override fun getId(): String = MOD_ID

}
